include/boost/url/grammar/impl/range_rule.hpp

```diff
@@ -372,9 +372,9 @@ class range<T>::
 
     iterator() = default;
     iterator(
-        iterator const&) noexcept = default;
+        iterator const&) = default;
```

Comment:
assignment of iterators is supposed to be noexcept, why the change?

Reply:
I removed that `noexcept` a loooooong time ago because I had an error somewhat similar to the error you had in http-proto later. I think some rule ended up not being `noexcept` and things failed but I couldn't find a solution for that.

I'll have another look. Maybe things fixed themselves. I don't know.

Reply:
Oh yes, because of `result<T>` assignment. We need to track that down and reproduce it with a minimal example, and see if Peter can fix it.

Reply:
we still need to track it down